### ISSUE TITLE
[FR] Provide static libc++ without exception

### ISSUE THREAD

**Issue by RikkaW** [android/ndk#1477]
**Is your feature request related to a problem? Please describe.**
Once using things that use exceptions, such as `new` and `unordered_set`, the exception will be included, which means a very huge size increase (~200KB per so file).

It is unacceptable the size of my so has increased from 14KB to 200KB only because I use `new`. Many things in the C++ world uses exception inside, it's impossible to avoid exceptions if we want to use C++. The option `-fno-exceptions` will not work of course since exceptions are included in the static library libc++ rather than the user's codes.

Crashing is so common in Android, and most times exception does not help. Use `new` as the example, if `new` (malloc inside) fails, it's obvious that throw an exception here does nothing help.

**Describe the solution you'd like**
Provide libcxx compiled with `-D_LIBCPP_NO_EXCEPTIONS` and `-D_LIBCPP_NO_RTTI` (and compiled with libcxxabi `-D_LIBCXXABI_NO_EXCEPTIONS`), then provided an option to let the user choose whether to use this libcxx or not.

**Comment by yujincheng08** [comment]
Agree. It's definitely annoying if I just want to use `std::unordered_map` but it introduces 200K of exception code. And what's worse, I have to introduce 4 copies of them for different ABIs.

**Comment by DoDoENT** [comment]
I would actually expand this feature request with the [ability to build the libc++ on your own](https://github.com/android/ndk/issues/1534). Besides wanting to disable exceptions and RTTI, I would also like to have unstable ABI so I could get more efficient implementations of some STL containers (e.g. string and variant - see the linked issue for the example).

**Comment by vvb2060** [comment]
https://github.com/RikkaW/libcxx-prefab
